src/core/qgsvectorlayerutils.h

```diff
@@ -176,6 +176,16 @@ class CORE_EXPORT QgsVectorLayerUtils
                                    QgsFieldConstraints::ConstraintStrength strength = QgsFieldConstraints::ConstraintStrengthNotSet,
                                    QgsFieldConstraints::ConstraintOrigin origin = QgsFieldConstraints::ConstraintOriginNotSet );
 
+    /**
+     * Tests an attribute \a value for type compatibility, i.e. checks whether it can be converted
+     * to the \a destinationType.
+     * NULL values (and QVariant invalid values because they are usually converted to NULLs) are considered valid.
```

Comment:
```suggestion
     * Will always return true for NULL and invalid QVariants.
```

That should be easier to understand and avoids reasoning.